src/Module.php

```diff
@@ -107,12 +137,34 @@ public function createSignature(SetaPDF_Core_Reader_FilePath $tmpPath)
         $padesDigest = $padesModule->getDigest();
 
         $hashData = hash($padesDigest, $padesModule->getDataToSign($tmpPath), true);
-        $signatureValue = $this->client->signWithOtp(
-            $this->certificateId,
-            $this->signingAlgorithm,
-            $hashData,
-            $this->otp
-        );
+
+        if ( $this->signingMethod == 'sad' ) {
+            // Sign using SAD
+            if ($this->sad === null) {
+                throw new \BadMethodCallException('Missing SAD! Did you perform /authorize request before?');
```

Comment:
`throw new \BadMethodCallException('Missing SAD!');`

IMHO the question doesn't make any sense in the context of this class.